dapp/module/db/db.go

```diff
@@ -81,10 +81,11 @@ func (m *Module) Register(vm *duktape.Context) error {
 			context.Call(1)
 
 		}
-		dec := make(chan struct{}, 1)
-		throttlingFunc(dec)
-		//m.reqLim.Exec(throttlingFunc)
-
+		m.reqLim.Exec(throttlingFunc)
+		// @TODO find a more reliable way to wait for reqLim.Exec to finish execution rather than time.Sleep(1 * time.Second)
+		// If we don't sleep here, the context is no longer available to the throttling module which tries to execute throttlingFunc,
```

Comment:
so I actually think that if you pass the context to the function it should work. Because we will end up in situations were `throttlingFunc` is called a minute later and sicne the process is async the context will be gone.  

Maybe you this could work: 

```go
m.reqLim.Exec(func (ctx Context) {
  return func() {
    // logic here
  }
}(ctx))
```

I am not sure if it works. The main problem is that we need to keep is async. We could wait till execution is done but we then block the whole VM ....